src/StubExecutable/StubExecutable.cpp

```diff
@@ -40,12 +41,96 @@ wchar_t* FindOwnExecutableName()
 	return ret;
 }
 
+bool Is64BitMachine()
+{
+	SYSTEM_INFO si;
+	GetNativeSystemInfo(&si);
+
+	if (((si.wProcessorArchitecture & PROCESSOR_ARCHITECTURE_IA64) == PROCESSOR_ARCHITECTURE_IA64) ||
+		((si.wProcessorArchitecture & PROCESSOR_ARCHITECTURE_AMD64) == PROCESSOR_ARCHITECTURE_AMD64))
+	{
+		return true;
+	}
+	else
+	{
+		return false;
+	}
+}
+
+void DeleteDirectory(std::wstring path)
+{
+	int len = path.length() + 2; // required to set 2 nulls at end of argument to SHFileOperation.
+	WCHAR* tempdir = new WCHAR[len]{0};
+
+	try
+	{
+		path.copy(tempdir, path.length());
+
+		SHFILEOPSTRUCT file_op = {
+		   NULL,
+		   FO_DELETE,
+		   tempdir,
+		   NULL,
+		   FOF_NOCONFIRMATION |
+		   FOF_NOERRORUI |
+		   FOF_SILENT,
+		   false,
+		   0,
+		   NULL };
+		SHFileOperation(&file_op);
+	}
+	catch (exception e) { }
+
+	delete[] tempdir;
+}
+
+void DeleteUnsupportedDirs(std::wstring searchPattern)
+{
+	//searchPattern: C:\Users\xxx\AppData\Local\UiPath\app-*
+	WIN32_FIND_DATA fileInfo = { 0 };
+	HANDLE hFile = FindFirstFile(searchPattern.c_str(), &fileInfo);
+	if (hFile == INVALID_HANDLE_VALUE) {
+		return;
+	}
+
+	version::Semver200_version acc("21.0.0");
+
+	do {
+		std::wstring appVer = fileInfo.cFileName;
+		if (!(fileInfo.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)) {
+			continue;
+		}
+		std::wstring appVerPath = searchPattern.substr(0, searchPattern.length() - 5) + appVer;
+		appVer = appVer.substr(4);   // Skip 'app-'
+
+		std::string s(appVer.begin(), appVer.end());
```

Comment:
I think you could simplify this logic a bit if the function took the following arguments:
`DeleteUnsupportedDirs(const wstring& directoryPath, const wstring& searchPattern)`

Then this code becomes `std::wstring appVer = filename.erase(0, searchPattern.size())`